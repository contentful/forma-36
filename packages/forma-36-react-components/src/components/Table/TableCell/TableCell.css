@import 'settings/colors';
@import 'settings/dimensions';
@import 'settings/typography';

.TableCell--head {
  background-color: var(--color-element-lightest);
  font-size: var(--font-size-s);
  font-weight: var(--font-weight-medium);
}

.TableCell--head__sorting {
  color: var(--color-text-dark);
}

.TableCell {
  border-bottom: 1px solid var(--color-element-light);
  color: var(--color-text-mid);
  font-size: var(--font-size-m);
  font-family: var(--font-stack-primary);
  font-weight: var(--font-weight-normal);
  line-height: 1.5;
  padding: var(--spacing-s) var(--spacing-m);
  vertical-align: top;
}

tr:hover td:not(.TableCell--head) {
  background-color: var(--color-element-lightest);
}

.TableRow--selected .TableCell {
  background-color: var(--color-blue-lightest);
}

.TableRow--selected .TableCell:last-child {
  box-shadow: inset -2px 0 0 var(--color-blue-mid);
}

tbody tr:last-child .TableCell {
  border-bottom: none;
}
