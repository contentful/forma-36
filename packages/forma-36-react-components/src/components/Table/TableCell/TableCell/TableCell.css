@import 'settings/colors';
@import 'settings/dimensions';
@import 'settings/typography';

.TableCell {
  border-bottom: 1px solid var(--color-element-light);
  color: var(--color-text-mid);
  font-size: var(--font-size-m);
  font-family: var(--font-stack-primary);
  font-weight: var(--font-weight-normal);
  line-height: 1.5;
  padding: var(--spacing-s) var(--spacing-m);
  vertical-align: top;
}

tbody tr:last-child .TableCell {
  border-bottom: none;
}

.TableCell--head {
  background-color: var(--color-element-lightest);
  font-size: var(--font-size-s);
  font-weight: var(--font-weight-medium);
}

.TableCell--head__sorting {
  color: var(--color-text-dark);
}
