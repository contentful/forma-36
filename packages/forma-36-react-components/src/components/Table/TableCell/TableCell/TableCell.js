import React from 'react';
import PropTypes from 'prop-types';
import cn from 'classnames';

import { TableCellContext } from './TableCellContext';

import styles from './TableCell.css';

export const sortingDirections = {
  asc: 'asc',
  desc: 'desc',
};

export class TableCell extends React.Component {
  static propTypes = {
    extraClassNames: PropTypes.string,
    children: PropTypes.node,
    sorting: PropTypes.oneOf([false, ...Object.keys(sortingDirections)]),
    align: PropTypes.oneOf(['center', 'left', 'right']),
  };

  static defaultProps = {
    children: null,
    extraClassNames: '',
    sorting: false,
    align: 'left',
  };

  render() {
    const {
      extraClassNames,
      children,
      sorting,
      align,
      ...otherProps
    } = this.props;

    return (
      <TableCellContext.Consumer>
        {({ name: context, element: Element, offsetTop }) => (
          <Element
            className={cn(styles.TableCell, extraClassNames, {
              [styles['TableCell--head']]: context === 'head',
              [styles['TableCell--head__sorting']]: sorting,
            })}
            style={{
              top: offsetTop,
            }}
            align={align}
            {...otherProps}
          >
            {children}
          </Element>
        )}
      </TableCellContext.Consumer>
    );
  }
}

export default TableCell;
