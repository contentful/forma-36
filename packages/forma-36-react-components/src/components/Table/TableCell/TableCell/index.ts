export { default } from './TableCell';
export { TableCellContext, contextOptions } from './TableCellContext';
