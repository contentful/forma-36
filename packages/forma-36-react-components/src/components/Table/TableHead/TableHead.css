@import 'settings/z-index';

.TableHead--sticky th {
  position: sticky;
  top: 0;
  z-index: var(--z-index-default);
}
