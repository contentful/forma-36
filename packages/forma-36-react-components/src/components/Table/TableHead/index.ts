export * from './TableHead';
export { default } from './TableHead';
