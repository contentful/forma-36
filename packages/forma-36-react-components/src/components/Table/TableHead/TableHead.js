import React from 'react';
import PropTypes from 'prop-types';
import cn from 'classnames';
import styles from './TableHead.css';

import { TableCellContext, contextOptions } from '../TableCell';

export class TableHead extends React.Component {
  static propTypes = {
    extraClassNames: PropTypes.string,
    children: PropTypes.node.isRequired,
    isSticky: PropTypes.bool,
    offsetTop: PropTypes.string,
  };

  static defaultProps = {
    extraClassNames: '',
    isSticky: false,
    offsetTop: undefined,
  };

  render() {
    const { extraClassNames, offsetTop, isSticky, children } = this.props;

    const classNames = cn(extraClassNames, {
      [styles[`TableHead--sticky`]]: isSticky,
    });
    return (
      <TableCellContext.Provider value={{ ...contextOptions.head, offsetTop }}>
        <thead className={classNames}>{children}</thead>
      </TableCellContext.Provider>
    );
  }
}

export default TableHead;
