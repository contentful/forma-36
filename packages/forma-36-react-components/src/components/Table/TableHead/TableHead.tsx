import React, { Component } from 'react';
import cn from 'classnames';
import styles from './TableHead.css';

import { TableCellContext, contextOptions } from '../TableCell';

export interface TableHeadProps {
  extraClassNames?: string;
  isSticky?: boolean;
  offsetTop?: string | number;
  children: React.ReactNode;
}

export class TableHead extends Component<TableHeadProps> {
  static defaultProps = {
    isSticky: false,
  };

  render() {
    const { extraClassNames, offsetTop, isSticky, children } = this.props;

    const classNames = cn(extraClassNames, {
      [styles[`TableHead--sticky`]]: isSticky,
    });
    return (
      <TableCellContext.Provider value={{ ...contextOptions.head, offsetTop }}>
        <thead className={classNames}>{children}</thead>
      </TableCellContext.Provider>
    );
  }
}

export default TableHead;
