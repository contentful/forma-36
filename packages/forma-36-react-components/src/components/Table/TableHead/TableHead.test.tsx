import React from 'react';
import { shallow } from 'enzyme';
import TableHead from './TableHead';
import TableCell from '../TableCell';

it('renders the component', () => {
  const output = shallow(
    <TableHead>
      <TableCell>test</TableCell>
    </TableHead>,
  );

  expect(output).toMatchSnapshot();
});

it('renders the component with additional className', () => {
  const output = shallow(
    <TableHead extraClassNames="extra-class-name">
      <TableCell>test</TableCell>
    </TableHead>,
  );

  expect(output).toMatchSnapshot();
});

it('renders the component as sticky', () => {
  const output = shallow(
    <TableHead isSticky>
      <TableCell>test</TableCell>
    </TableHead>,
  );

  expect(output).toMatchSnapshot();
});

it('renders the component as sticky and with an offset Top', () => {
  const output = shallow(
    <TableHead isSticky offsetTop="20px">
      <TableCell>test</TableCell>
    </TableHead>,
  );

  expect(output).toMatchSnapshot();
});
