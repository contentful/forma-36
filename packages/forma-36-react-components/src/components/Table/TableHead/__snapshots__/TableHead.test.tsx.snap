// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the component 1`] = `
<ContextProvider
  value={
    Object {
      "element": "th",
      "name": "head",
      "offsetTop": undefined,
    }
  }
>
  <thead
    className=""
  >
    <TableCell
      align="left"
      sorting={false}
    >
      test
    </TableCell>
  </thead>
</ContextProvider>
`;

exports[`renders the component as sticky 1`] = `
<ContextProvider
  value={
    Object {
      "element": "th",
      "name": "head",
      "offsetTop": undefined,
    }
  }
>
  <thead
    className="TableHead--sticky"
  >
    <TableCell
      align="left"
      sorting={false}
    >
      test
    </TableCell>
  </thead>
</ContextProvider>
`;

exports[`renders the component as sticky and with an offset Top 1`] = `
<ContextProvider
  value={
    Object {
      "element": "th",
      "name": "head",
      "offsetTop": "20px",
    }
  }
>
  <thead
    className="TableHead--sticky"
  >
    <TableCell
      align="left"
      sorting={false}
    >
      test
    </TableCell>
  </thead>
</ContextProvider>
`;

exports[`renders the component with additional className 1`] = `
<ContextProvider
  value={
    Object {
      "element": "th",
      "name": "head",
      "offsetTop": undefined,
    }
  }
>
  <thead
    className="extra-class-name"
  >
    <TableCell
      align="left"
      sorting={false}
    >
      test
    </TableCell>
  </thead>
</ContextProvider>
`;
